src/test/java/it/pagopa/pn/bff/pnclient/userattributes/PnUserAttributesClientImplTestIT.java

```diff
@@ -142,4 +143,186 @@ void acceptTosConsentError() throws JsonProcessingException {
+                );
+
+        StepVerifier.create(pnUserAttributesClient.createOrUpdateLegalAddress(
+                UserMock.PN_UID,
+                CX_TYPE,
+                UserMock.SENDER_ID,
+                LegalChannelType.PEC,
+                addressesMock.getAddressVerificationBodyMock(),
+                UserMock.PN_CX_GROUPS,
+                UserMock.PN_CX_ROLE
+        )).expectNext(addressesMock.addressVerificationCourtesyResponseMock()).verifyComplete();
+    }
+
+    @Test
+    void createOrUpdateLegalAddressError() throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String request = objectMapper.writeValueAsString(addressesMock.getAddressVerificationBodyMock());
+        mockServerClient.when(request().withMethod("POST").withPath(addressPath + "/legal/default/PEC").withBody(request))
+                .respond(response().withStatusCode(404));
+
+        StepVerifier.create(pnUserAttributesClient.createOrUpdateLegalAddress(
+                UserMock.PN_UID,
+                CX_TYPE,
+                UserMock.SENDER_ID,
+                LegalChannelType.PEC,
+                addressesMock.getAddressVerificationBodyMock(),
+                UserMock.PN_CX_GROUPS,
+                UserMock.PN_CX_ROLE
+        )).expectError().verify();
+    }
+
+    @Test
+    void deleteCourtesyAddress() {
+        mockServerClient.when(request().withMethod("DELETE").withPath(addressPath + "/courtesy/default/EMAIL"))
+                .respond(response()
+                        .withStatusCode(200)
+                        .withContentType(MediaType.APPLICATION_JSON)
+                        .withBody("")
+                );
+
+        StepVerifier.create(pnUserAttributesClient.deleteRecipientCourtesyAddress(
+                UserMock.PN_UID,
+                CX_TYPE,
+                UserMock.SENDER_ID,
+                CourtesyChannelType.EMAIL,
+                UserMock.PN_CX_GROUPS,
+                UserMock.PN_CX_ROLE
+        )).expectNext().verifyComplete();
+    }
+
+    @Test
+    void deleteCourtesyAddressError() {
+        mockServerClient.when(request().withMethod("DELETE").withPath(addressPath + "/courtesy/default/EMAIL"))
+                .respond(response().withStatusCode(404));
+
+        StepVerifier.create(pnUserAttributesClient.deleteRecipientCourtesyAddress(
+                UserMock.PN_UID,
+                CX_TYPE,
+                UserMock.SENDER_ID,
+                CourtesyChannelType.EMAIL,
+                UserMock.PN_CX_GROUPS,
+                UserMock.PN_CX_ROLE
+        )).expectError().verify();
+    }
+
+    @Test
+    void deleteLegalAddress() {
+        mockServerClient.when(request().withMethod("DELETE").withPath(addressPath + "/legal/default/PEC"))
+                .respond(response()
+                        .withStatusCode(200)
+                        .withContentType(MediaType.APPLICATION_JSON)
+                        .withBody("")
```

Comment:
Here we can remove the .withContentType(MediaType.APPLICATION_JSON) and .withBody("")